pkg/keystore/provider.go

```diff
@@ -6,8 +6,32 @@ SPDX-License-Identifier: Apache-2.0
 
 package keystore
 
+import (
+	"errors"
+
+	"github.com/trustbloc/edge-core/pkg/storage"
+)
+
+// ErrDuplicateKeystore is returned when an attempt is made to create a duplicate keystore.
+var ErrDuplicateKeystore = errors.New("duplicate keystore")
+
+// Configuration represents a keystore configuration.
+type Configuration struct {
+	Sequence   int    `json:"sequence"`
+	Controller string `json:"controller"`
+}
+
 // Provider represents functionality needed for keystore.
 type Provider interface {
 	// CreateStore creates a new keystore with the given name.
 	CreateStore(name string) error
+
+	// OpenStore opens an existing store and returns it.
+	OpenStore(name string) (storage.Store, error)
+}
+
+// Store represents underlying storage for keystore.
+type Store interface {
```

Comment:
Can't we directly use store provider from [edge-core](https://github.com/trustbloc/edge-core/blob/master/pkg/storage/store.go) ?

For me, the keystore implementation looks similar to profile store in [vc-services](https://github.com/trustbloc/edge-service/blob/master/pkg/doc/vc/profile/verifier/profile.go#L37). We can have APIs like Create, Update, delete Keystore and have custom logic/validations related to it with store provider passed to the constructor.

Reply:
You are right, it's not necessary to have that interface (at least with current requirements to the underlying storage for a keystore). Thank you Rolson for catching that.